packages/accounts-base/accounts_server.js

```diff
@@ -1135,13 +1135,15 @@ Ap._expirePasswordResetTokens = function (oldestValidDate, userId) {
   var userFilter = userId ? {_id: userId} : {};
 
   this.users.update(_.extend(userFilter, {
+    "services.password.reset.reason": reason,
     $or: [
       { "services.password.reset.when": { $lt: oldestValidDate } },
       { "services.password.reset.when": { $lt: +oldestValidDate } }
     ]
   }), {
     $unset: {
       "services.password.reset": {
```

Comment:
$unset is not changed in this PR. I have just updated the query. Here is the documentation for $unset
https://docs.mongodb.com/manual/reference/operator/update/unset/